### ISSUE TITLE
ardb hanging for long intervals after DEL of a very large list

### ISSUE THREAD

**Issue by bernardko** [yinqiwen/ardb#406]
Hi @yinqiwen, I have a very large list around 800M strings which I deleted using the DEL command. 

After deleting it, commands that I think scan the key space causes ardb to hang for long periods of time for around one hour while running the command. I tried "KEYS *" and "BGSAVE" which both cause hanging. While it is hanging I noticed that ardb was doing alot of reading and memory increased to 6GB. After around 1 hour of hanging, the command completes. The memory doesn't increase much more after around 6GB even when running the commands again.

I also noticed that the ardb database file size didn't hasn't changed at around 87GB.

After I ran "RENAME" on another list to the same name as the one I deleted and that took 300 seconds to complete and I noticed that rocksdb background task writing to disk. Now the files at around 82GB. I then ran "KEY *" and now that command takes 14 seconds to run. My database only has 14 keys. "SCAN 0" also takes around the same time. "BGSAVE" still takes a very long time to finish, I noticed that the snapshot reaches the its full size and then hangs for an hour or so before the BGSAVE is finished.

Also just for your info, I'm running on the newest commit using rocksdb with default settings. This particular database is running on a spinning disk.

Do you have any insight on what is happening here? Thanks for your help!

**Comment by yinqiwen** [comment]
delete a large list use rocksdb's experiment feature [range delete](https://github.com/facebook/rocksdb/wiki/Delete-A-Range-Of-Keys).  It's normal that cost a long time to scan whole database since the complexity is O(N) while N contains deleted values.   the deleted values would be discard by background compact task of rocksdb later.

u can use `compactall` to discard deleted values in rocksdb

**Comment by bernardko** [comment]
Thanks for the hint. I noticed the ```compactall``` command in your docs but it says:

```CompactAll - Used to compact disk store(only work for LevlDB engine) for all range.```

So I thought it wouldn't work for rocksdb, thanks for that info. I will try it out.

I also ran into a performance issue in another database that has a workload which deletes alot of keys. After deleting alot of smaller sized keys over a very long period of time, the performance of the commands dropped dramatically, taking 1 second or more per call. That seems related to this as well, I'll try running ```compactall``` on that DB to test it out.
